### ISSUE TITLE
Safari: can't write in composer (redactor or quill)

### ISSUE THREAD

**Issue by whuut** [NodeBB/NodeBB#8867]
- **NodeBB version:** 1.15.0
- **NodeBB git hash:** a30aa5629cb51c881ea99431e739c8813ecfded8

- **Installed NodeBB Plugins:** 
Active plugins:
        * nodebb-plugin-composer-default (installed, disabled)
        * nodebb-plugin-composer-redactor (installed, enabled)
        * nodebb-plugin-custom-file-extensions (not installed, enabled)
        * nodebb-plugin-dbsearch (installed, enabled)
        * nodebb-plugin-emoji (installed, disabled)
        * nodebb-plugin-emoji-android (installed, disabled)
        * nodebb-plugin-emoji-extended (installed, enabled)
        * nodebb-plugin-emoji-one (installed, enabled)
        * nodebb-plugin-markdown (installed, enabled)
        * nodebb-plugin-mentions (installed, enabled)
        * nodebb-plugin-newuser-approval (installed, enabled)
        * nodebb-plugin-soundpack-default (installed, disabled)
        * nodebb-plugin-spam-be-gone (installed, enabled)
        * nodebb-rewards-essentials (installed, enabled)
        * nodebb-theme-lavender (installed, disabled)
        * nodebb-theme-persona (installed, enabled)
        * nodebb-theme-slick (installed, disabled)
        * nodebb-theme-vanilla (installed, disabled)
        * nodebb-widget-essentials (installed, enabled)

- **Database type:** mongo
- **Database version:** v3.6.20

- **Exact steps to cause this issue:**

We upgraded from version 1.14.3 to 1.15.0 and since then all our Safari users can't write a post anymore.

Edit: When you click reply, the composer window opens, but there is no cursor and you can't write anything.
Edit2: it works with Chrome and Firefox on the same machines though.

We are using composer-redactor and since it seems like it has been abandoned we thought we would try composer-quill on a test installation. But it happens there too.

There are no errors in the development console.

We tried rebuilding assets and restarting multiple times, but it did not help.
All users cleared their browser cache too.

The output log while rebuilding says:

```
2020-11-11T12:13:00.924Z [4569/2040] - ESC[32minfoESC[39m: [app] Database connection closed.
2020-11-11T12:13:00.927Z [4569/2040] - ESC[32minfoESC[39m: [app] Shutdown complete.
[cluster] Child Process (2040) has exited (code: 0, signal: null)
2020-11-11T12:13:01.301Z [4569/2512] - ESC[32minfoESC[39m: Initializing NodeBB v1.15.0 https://domain.com
[benchpressjs] Unable to build or find a suitable native module, falling back to JS version.
[benchpressjs] This will result in much longer template build times.
[benchpressjs] prebuilt module failed to load: linux_72
[benchpressjs] If the above module is not available at `node_modules/benchpressjs/rust/benchpress-rs/pre-built/`,
[benchpressjs] you can manually build benchpress-rs following the instructions here:
[benchpressjs] `https://github.com/benchpressjs/benchpressjs/blob/master/README.md#manually-building-native-module`
2020-11-11T12:13:03.291Z [4569/2512] - ESC[33mwarnESC[39m: [upgrade/appendPluginScripts] Unable to read plugin.json for plugin `nodebb-plugin-custom-file-extensions`. Skipping.
2020-11-11T12:13:03.581Z [4569/2512] - ESC[32minfoESC[39m: [socket.io] Restricting access to origin: https://domain.com:*
2020-11-11T12:13:04.378Z [4569/2512] - ESC[33mwarnESC[39m:    [plugins/nodebb-theme-persona] The plugin.json field "library" is deprecated. Please use the package.json field "main" instead.
2020-11-11T12:13:04.609Z [4569/2512] - ESC[33mwarnESC[39m:    [plugins/nodebb-plugin-spam-be-gone] The plugin.json field "library" is deprecated. Please use the package.json field "main" instead.
2020-11-11T12:13:04.670Z [4569/2512] - ESC[33mwarnESC[39m:    [plugins/nodebb-plugin-newuser-approval] The plugin.json field "library" is deprecated. Please use the package.json field "main" instead.
2020-11-11T12:13:04.671Z [4569/2512] - ESC[33mwarnESC[39m: [plugins] Unable to load library for: nodebb-plugin-newuser-approval
2020-11-11T12:13:04.671Z [4569/2512] - ESC[31merrorESC[39m: Error: Cannot find module './meta'
Require stack:
- /var/lib/nodebb/src/plugins/index.js
- /var/lib/nodebb/src/meta/configs.js
- /var/lib/nodebb/src/meta/index.js
- /var/lib/nodebb/src/start.js
- /var/lib/nodebb/require-main.js
- /var/lib/nodebb/app.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:15)
    at Function.Module._load (internal/modules/cjs/loader.js:842:27)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
    at Object.<anonymous> (/var/lib/nodebb/node_modules/nodebb-plugin-newuser-approval/library.js:4:26)
    at Module._compile (internal/modules/cjs/loader.js:1138:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1158:10)
    at Module.load (internal/modules/cjs/loader.js:986:32)
    at Function.Module._load (internal/modules/cjs/loader.js:879:14)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
    at require (internal/modules/cjs/helpers.js:72:18)
2020-11-11T12:13:04.778Z [4569/2512] - ESC[32minfoESC[39m: [plugins/spam-be-gone] Settings loaded
2020-11-11T12:13:04.796Z [4569/2512] - ESC[32minfoESC[39m: [api] Adding 0 route(s) to `api/v3/plugins`
2020-11-11T12:13:04.813Z [4569/2512] - ESC[32minfoESC[39m: Routes added
2020-11-11T12:13:04.817Z [4569/2512] - ESC[32minfoESC[39m: NodeBB Ready
2020-11-11T12:13:04.819Z [4569/2512] - ESC[32minfoESC[39m: Enabling 'trust proxy'
2020-11-11T12:13:04.821Z [4569/2512] - ESC[32minfoESC[39m: NodeBB is now listening on: 127.0.0.1:4569
```

But it seems like this is not composer related.

Can anyone have a look at this please?

Thank you and best regards,
whuut

**Comment by julianlam** [comment]
Hi there,

Unfortunately Salari is one of our blind spots since we do not have any developers using it as their daily driver Can you please let us know what the client-side errors say in the dev console, when you attempt to open the composer?

**Comment by whuut** [comment]
Hi @julianlam,

the only messages that appeared were some 404 not found on some .js.map files

At first it was only these 2:

```
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (pulling-drawer.js.map, line 0)
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (jquery.form.min.js.map, line 0)
```

I thought they were not relevant because I got them on Chrome or Firefox too and it worked there.
But I copied them anyways:

```
cp /var/lib/nodebb/node_modules/pulling/build/pulling-drawer.js.map /var/lib/nodebb/public/src/modules/
cp /var/lib/nodebb/node_modules/jquery-form/dist/jquery.form.min.js.map /var/lib/nodebb/public/src/modules/
```

But now while using composer-quill there are new errors:
```
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (index.js.map, line 0)
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (emoji.js.map, line 0)
```

But this time I don't know what to copy because there are more than one instances of `index.js.map` for example.

Shouldn't the `./nodebb upgrade` or asset rebuild have taken care of this?


Best regards,
whuut

**Comment by whuut** [comment]
On a side note: on our prodction forum where we still use composer-redactor, there are no more error messages in the console (after I copied the first 2 files above) and the problem still perists. So I'm wondering if these map files are even related to this issue.

**Comment by julianlam** [comment]
They're not :smile: The .map files are useful from a debugging perspective but do not affect anything functionality-wise. Sadly those errors don't help, but I will have to figure it out with the help of something like Browserstack or LamdaTest, will get back to you on this...

**Comment by julianlam** [comment]
Fixed via 711c094f9b760e6148741dce66d2d5d0256ce469, pending confirmation
